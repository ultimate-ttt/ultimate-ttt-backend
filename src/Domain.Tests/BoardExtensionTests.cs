using FluentAssertions;
using Snapshooter.Xunit;
using UltimateTicTacToe.Abstractions;
using Xunit;

namespace UltimateTicTacToe.Domain.Tests
{
    public class BoardExtensionTests
    {
        [Fact]
        public void GetAllLinesOfBoard_ValidSmallTileInformation_ShouldReturnAllLines()
        {
            // arrange
            var board = new SmallTileInformation[3][];

            for (int x = 0; x < 3; x++)
            {
                board[x] = new SmallTileInformation[3];
                for (int y = 0; y < 3; y++)
                {
                    board[x][y] = new SmallTileInformation
                    {
                        BoardPosition = new Position(0, 0),
                        Position = new Position(x, y),
                        Value = TileValue.Empty,
                    };
                }
            }

            // act
            var result = board.GetAllLinesOfBoard();

            // assert
            result.Should().HaveCount(8);
            Snapshot.Match(result);
        }

        [Fact]
        public void GetAllLinesOfBoard_ValidSmallBoardInformation_ShouldReturnAllLines()
        {
            // arrange
            var board = new SmallBoardInformation[3][];

            for (int x = 0; x < 3; x++)
            {
                board[x] = new SmallBoardInformation[3];
                for (int y = 0; y < 3; y++)
                {
                    board[x][y] = new SmallBoardInformation
                    {
                       Position = new Position(x, y),
                       Value = TileValue.Empty,
                    };
                }
            }

            // act
            var result = board.GetAllLinesOfBoard();

            // assert
            result.Should().HaveCount(8);
            Snapshot.Match(result);
        }
    }
}
