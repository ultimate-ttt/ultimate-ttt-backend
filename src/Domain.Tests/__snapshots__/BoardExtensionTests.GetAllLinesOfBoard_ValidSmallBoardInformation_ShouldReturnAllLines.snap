[
  [
    {
      "Tiles": null,
      "Position": {
        "X": 0,
        "Y": 0
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 0,
        "Y": 1
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 0,
        "Y": 2
      },
      "Value": "Empty"
    }
  ],
  [
    {
      "Tiles": null,
      "Position": {
        "X": 0,
        "Y": 0
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 1,
        "Y": 0
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 2,
        "Y": 0
      },
      "Value": "Empty"
    }
  ],
  [
    {
      "Tiles": null,
      "Position": {
        "X": 1,
        "Y": 0
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 1,
        "Y": 1
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 1,
        "Y": 2
      },
      "Value": "Empty"
    }
  ],
  [
    {
      "Tiles": null,
      "Position": {
        "X": 0,
        "Y": 1
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 1,
        "Y": 1
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 2,
        "Y": 1
      },
      "Value": "Empty"
    }
  ],
  [
    {
      "Tiles": null,
      "Position": {
        "X": 2,
        "Y": 0
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 2,
        "Y": 1
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 2,
        "Y": 2
      },
      "Value": "Empty"
    }
  ],
  [
    {
      "Tiles": null,
      "Position": {
        "X": 0,
        "Y": 2
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 1,
        "Y": 2
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 2,
        "Y": 2
      },
      "Value": "Empty"
    }
  ],
  [
    {
      "Tiles": null,
      "Position": {
        "X": 0,
        "Y": 0
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 1,
        "Y": 1
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 2,
        "Y": 2
      },
      "Value": "Empty"
    }
  ],
  [
    {
      "Tiles": null,
      "Position": {
        "X": 2,
        "Y": 0
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 1,
        "Y": 1
      },
      "Value": "Empty"
    },
    {
      "Tiles": null,
      "Position": {
        "X": 0,
        "Y": 2
      },
      "Value": "Empty"
    }
  ]
]
